src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -110,44 +121,126 @@ protected override void Dispose(bool disposing)
 
         #region Private methods
 
-        private void StartRequest(RequestCompletionSource state)
+        private async void StartRequest(RequestCompletionSource state)
```

Comment:
The handlers for HttpClient NEVER buffer response entity-body.  The design principle is that the .SendAsync() method of a handler returns an `HttpResponseMessage` object with `HttpResponseMessage.Content` property as an `StreamContent` object.

The higher layer APIs in `HttpClient` for example can choose whether a particular overload of `SendAsync` will auto-buffer response body.  Or developers can also buffer the response body via the `HttpContent.LoadIntoBufferAsync` APIs.